<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>serialPortWidget</class>
 <widget class="QWidget" name="serialPortWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>409</width>
    <height>407</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Serial Port</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_4">
   <item row="0" column="0">
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="spacing">
      <number>1</number>
     </property>
     <property name="sizeConstraint">
      <enum>QLayout::SetFixedSize</enum>
     </property>
     <item>
      <widget class="QPushButton" name="showSerialPortButton">
       <property name="text">
        <string>Serial Port</string>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
       <property name="autoDefault">
        <bool>false</bool>
       </property>
       <property name="default">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QWidget" name="portConfigWidget" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <layout class="QGridLayout" name="gridLayout">
        <item row="2" column="1">
         <widget class="QLabel" name="timeMessageLabel">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>port closed</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Time since last message:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QGroupBox" name="selectBox">
          <property name="title">
           <string>Serial Port Info</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_3">
           <item row="3" column="0">
            <widget class="QLabel" name="locationLabel">
             <property name="text">
              <string>Location:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="descriptionLabel">
             <property name="text">
              <string>Description:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="manufacturerLabel">
             <property name="text">
              <string>Manufacturer:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="serialNumberLabel">
             <property name="text">
              <string>Serial number:</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QGroupBox" name="parametersBox">
          <property name="title">
           <string>Select Parameters</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="1" column="1">
            <widget class="QComboBox" name="dataBitsBox"/>
           </item>
           <item row="3" column="1">
            <widget class="QComboBox" name="stopBitsBox"/>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="stopBitsLabel">
             <property name="text">
              <string>Stop bits:</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QComboBox" name="flowControlBox"/>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="flowControlLabel">
             <property name="text">
              <string>Flow control:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QComboBox" name="parityBox"/>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="baudRateBox"/>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="baudRateLabel">
             <property name="text">
              <string>BaudRate:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="dataBitsLabel">
             <property name="text">
              <string>Data bits:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="parityLabel">
             <property name="text">
              <string>Parity:</string>
             </property>
            </widget>
           </item>
           <item row="5" column="0" colspan="2">
            <widget class="QCheckBox" name="enableLogCheckbox">
             <property name="text">
              <string>Enable Log</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="3" column="0" colspan="2">
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QPushButton" name="openPortButton">
            <property name="text">
             <string>Open Port</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="closePortButton">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Close Port</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
